src/exoticatechnologies/hullmods/exotics/ExoticHullmod.kt

```diff
@@ -0,0 +1,30 @@
+package exoticatechnologies.hullmods.exotics
+
+import com.fs.starfarer.api.combat.BaseHullMod
+import com.fs.starfarer.api.combat.MutableShipStatsAPI
+import com.fs.starfarer.api.combat.ShipAPI
+import com.fs.starfarer.api.combat.ShipAPI.HullSize
+import exoticatechnologies.modifications.exotics.impl.HullmodExotic
+
+/**
+ * Class that should be used exclusively by "exotic hullmods" used by [HullmodExotic]s.
+ *
+ * It's a slight extension on top of [BaseHullMod] that should let us cleanup after the hullmod has been removed.
+ */
+abstract class ExoticHullmod: BaseHullMod() {
+
+    /**
+     * This needs to match the ID from the hull_mods.csv
+     */
+    abstract val hullModId: String
+
+    /**
+     * You should undo all effects you've applied to the [stats] in [applyEffectsBeforeShipCreation]
+     */
+    abstract fun removeEffectsBeforeShipCreation(hullSize: HullSize, stats: MutableShipStatsAPI, id: String)
+
+    /**
+     * You should undo all effects you've applied to [ship] in [applyEffectsAfterShipCreation]
+     */
+//    abstract fun removeEffectsAfterShipCreation(ship: ShipAPI, id: String)
```

Comment:
get rid of this. maybe add later if/when it's necessary.

but I don't think it's necessary at all, because all modifications made to `ShipAPI` can be handled **just fine** by Exotic::onDestroy()

Reply:
lets rethink this later on. for now, it's commented out and javadocced properly :D

Reply:
nah, removed it. done.

Reply:
resolving.